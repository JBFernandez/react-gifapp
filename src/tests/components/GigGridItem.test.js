import React from 'react';
import GifGridItem from '../../components/GifGridItem';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';

describe('Tests on <GifGridItem/>', () => { 

    const title = 'A title';
    const url = 'https:localhost/example.jpg';
    const wrapper = shallow( <GifGridItem title={title} url={url} /> );

    test('should display GifGridItem correctly', () => { 


        expect(wrapper).toMatchSnapshot();
        
     })//test

     test('should have a paragraph with the title', () => { 
         const p = wrapper.find('p');
         expect( p.text().trim() ).toBe(title);
      })//test paragraph

      test('img should be equal to url and alt of props', () => { 
          const img = wrapper.find('img');
          //console.log( img.prop('src') );
          expect(img.prop('src')).toBe(url);
          expect(img.prop('alt')).toBe(title);
       })//test img props

       test('should have animate_fadeIn', () => { 
           const div = wrapper.find('div');
           const className = div.prop('className');
           expect(className.includes('animate__fadeIn')).toBe(true);
           expect(className.includes('animate__fadeIn')).not.toBe(false);
        })

 })//describe
