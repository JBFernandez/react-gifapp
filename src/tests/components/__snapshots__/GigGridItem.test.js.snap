// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGridItem/> should display GifGridItem correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    src="https:localhost/example.jpg"
  />
  <p>
     
    A title
     
  </p>
</div>
`;
